Plutarch/Api/V1/AssocMap.hs

```diff
@@ -63,14 +64,24 @@ import Plutarch.Unsafe (punsafeCoerce, punsafeDowncast)
 
 import Prelude hiding (all, any, filter, lookup, null)
 
+import Data.Proxy (Proxy (Proxy))
+
 data KeyGuarantees = Sorted | Unsorted
 
 type role PMap nominal nominal nominal nominal
 newtype PMap (keysort :: KeyGuarantees) (k :: PType) (v :: PType) (s :: S) = PMap (Term s (PBuiltinMap k v))
   deriving stock (Generic)
-  deriving anyclass (PlutusType, PIsData, PEq, PShow)
+  deriving anyclass (PlutusType, PIsData, PShow)
 instance DerivePlutusType (PMap keysort k v) where type DPTStrat _ = PlutusTypeNewtype
 
+instance PEq (PMap 'Unsorted k v)
```

Comment:
I don't think this is correct. This instance should be removed, no?

Reply:
is that not how it was derived before for unsorted map? this is the same as anyclass deriving

Reply:
yep, and I'm pretty sure this is incorrect

Reply:
What do you mean? Like in you don't want two maps that are equal modulo sortedness to be seen as unequal? I think that's fine tbh.

Reply:
So you just want me to remove it? that means we don't have any notion of equality for Unsorted Maps?

Reply:
Yes, it's confusing.

Reply:
You can always just pto then compare.

Reply:
okay then, lemme remove it